tests/common.py

```diff
@@ -344,10 +344,13 @@ def _async_fire_mqtt_message(topic, payload, qos, retain):
         assert result
         await hass.async_block_till_done()
 
-        hass.data["mqtt"] = MagicMock(
+        mqtt_component_mock = MagicMock(
             spec_set=hass.data["mqtt"], wraps=hass.data["mqtt"]
         )
+        hass.data["mqtt"].connected = mqtt_component_mock.connected
+        mqtt_component_mock._mqttc = mock_client
```

Comment:
Can we break out the client in a separate fixture, which we depend on in this fixture, so we can assert things about the client without accessing a private attribute?

Reply:
Done!
I will follow up with another PR which replaces the use of `async_mock_mqtt_component` and `mock_mqtt_component` with using the fixtures.